src/store/drawing/__tests__/reducer-test.ts

```diff
@@ -63,11 +77,67 @@ describe('store/drawing/reducer', () => {
     describe('createAnnotationAction.fulfilled', () => {
         test('should reset to initial state', () => {
             const newState = reducer(
-                { location: 1, drawnPathGroups: [pathGroups[0]] },
+                { location: 1, drawnPathGroups: [pathGroup1], stashedPathGroups: [pathGroup2] },
                 createAnnotationAction.fulfilled,
             );
 
             expect(newState).toMatchObject(initialState);
         });
     });
+
+    describe('redoDrawingPathGroupAction', () => {
+        test('should do nothing if stashedPathGroups is empty', () => {
+            const newState = reducer(
+                { location: 1, drawnPathGroups: [pathGroup1], stashedPathGroups: [] },
+                redoDrawingPathGroupAction(),
+            );
+
+            expect(newState).toMatchObject({
+                drawnPathGroups: [pathGroup1],
+                location: 1,
+                stashedPathGroups: [],
+            });
+        });
+
+        test('should pop from stashedPathGroups and push onto drawnPathGroups', () => {
+            const newState = reducer(
+                { location: 1, drawnPathGroups: [pathGroup1], stashedPathGroups: [pathGroup2] },
+                redoDrawingPathGroupAction(),
+            );
+
+            expect(newState).toMatchObject({
+                drawnPathGroups: [pathGroup1, pathGroup2],
+                location: 1,
+                stashedPathGroups: [],
+            });
+        });
+    });
+
+    describe('undoDrawingPathGroupAction', () => {
+        test('should do nothing if drawnPathGroups is empty', () => {
+            const newState = reducer(
+                { location: 1, drawnPathGroups: [], stashedPathGroups: [] },
+                undoDrawingPathGroupAction(),
+            );
+
+            expect(newState).toMatchObject({
+                drawnPathGroups: [],
+                location: 1,
+                stashedPathGroups: [],
+            });
+        });
+
+        test('should pop from drawnPathGroups and push onto stashedPathGroups', () => {
+            const newState = reducer(
+                { location: 1, drawnPathGroups: [pathGroup1], stashedPathGroups: [pathGroup2] },
+                undoDrawingPathGroupAction(),
+            );
+
+            expect(newState).toMatchObject({
+                drawnPathGroups: [],
+                location: 1,
```

Comment:
Do we need to assert `location` in every test? Can it be affected by this action?